src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,30 @@
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User cannot be null");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login cannot be null");
+        }
+        if (user.getLogin().length() < 6) {
+            throw new RegistrationException("Login must be at least 6 characters");
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password cannot be null");
+        }
+        if (user.getPassword().length() < 6) {
+            throw new RegistrationException("Password must be at least 6 characters");
+        }
+        if (user.getAge() == null) {
+            throw new RegistrationException("Age cannot be null");
+        }
+        if (user.getAge() < 18) {
```

Comment:
The age check should also use a constant field for the minimum age value. This will make your code more maintainable and the constants' names will provide additional context.